test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/DeepLinkPackageParserTests.cs

```diff
@@ -0,0 +1,103 @@
+
+            string resultPackageName = resultingPackage.PackageName;
+            string resultPackageVersion = resultingPackage.VersionNumber;
+
+            Assert.NotNull(resultingPackage);
+            Assert.Equal(testPackageName, resultPackageName);
+            Assert.Equal(testPackageVersion, resultPackageVersion);
+        }
+
+        [Theory]
+        [InlineData("nuget-client://OpenPackageDetails/Newtonsoft.Json", "Newtonsoft.Json")]
+        [InlineData("nuget-client://OpenPackageDetails/Serilog", "Serilog")]
+        [InlineData("nuget-client://OpenPackageDetails/Moq", "Moq")]
+        public void URILinks_FormattedProperlyOneProperty_ShouldBeValid(string packageLink, string packageName)
+        {
+            //Arrange
+            var package = new NuGetPackageDetails(packageName);
+
+            //Act
+            NuGetPackageDetails resultingPackage = DeepLinkURIParser.GetNuGetPackageDetails(packageLink);
+
+            //Assert
+            string testPackageName = package.PackageName;
+
+            string resultPackageName = resultingPackage.PackageName;
+            string resultPackageVersion = resultingPackage.VersionNumber;
+
+            Assert.NotNull(resultingPackage);
+            Assert.Equal(testPackageName, resultPackageName);
+            Assert.Null(resultPackageVersion);
+        }
+
+        [Theory]
+        [InlineData("nuget-client:://OpenPackageDetails/Newtonsoft.Json/1.0.0")]
+        [InlineData("nuget-client:/a/OpenPackageDetails/Serilog/1.2.3")]
+        [InlineData("vsah://OpenPackageDetails/Moq/1.3.5")]
+        public void URILinks_BadProtocolName_ShouldBeNull(string packageLink)
+        {
+            //Arrange
+
+            //Act
+            NuGetPackageDetails result = DeepLinkURIParser.GetNuGetPackageDetails(packageLink);
+
+            //Assert
+            Assert.Null(result);
+        }
+
+        [Theory]
+        [InlineData("nuget-client://OpenSesame/Newtonsoft.Json/1.0.0")]
+        [InlineData("nuget-client://HEyHOwdidIgetHere/Serilog/1.2.3")]
+        [InlineData("nuget-client://OpenPackageDetail/Moq/1.3.5")]
+        public void URILinks_BadDomainName_ShouldBeNull(string packageLink)
+        {
+            //Arrange
+
+            //Act
+            NuGetPackageDetails result = DeepLinkURIParser.GetNuGetPackageDetails(packageLink);
+
+            //Assert
+            Assert.Null(result);
+        }
+
+        [Theory]
+        [InlineData("nuget-client://OpenPackageDetails/Newtsoft.Json/1.0.0/wazzap")]
+        [InlineData("nuget-client://OpenPackageDetails/Serilog/1.2.3/Im/messing/up/your/code")]
+        [InlineData(null)]
+        public void URILinks_NumberOfPropertiesIsNotTwoOrOne_ShouldBeNull(string packageLink)
+        {
+            //Arrange
+
+            //Act
+            NuGetPackageDetails result = DeepLinkURIParser.GetNuGetPackageDetails(packageLink);
+
+            //Assert
+            Assert.Null(result);
+        }
```

Comment:
Please add a test for when the version is supplied but invalid.

What is the behavior in this scenario? Do we just ignore the version or do we consider the whole thing invalid? Whatever the behavior, make sure the test reflects this and so does the code.

Reply:
This is only testing the link to see if it has all the appropriate sections, it does not do any work in validating the version